### PR TITLE
Bump kubebuilder-presubmits to use go 1.18

### PR DESCRIPTION
**Pull request by AlmogBaku**
*Description:*
Kubebuilder's master branch is introducing support for Golang 1.18. This PR is changing the `pull-kubebuilder-test` pre-submit job to use `golang:1.18` image instead of the previous version.

*Motivation:*
Kubebuilder Golang 1.18 Support PR https://github.com/kubernetes-sigs/kubebuilder/pull/2561
Kubebuilder Golang 1.18 Issue https://github.com/kubernetes-sigs/kubebuilder/issues/2559

### CODE REVIEW

**Comment by AlmogBaku** [comment]
@camilamacedo86 can you please add lgtm?

**Comment by camilamacedo86** [comment]
Hi @AlmogBaku,  

It is great that we have this PR open already. We will need it.
However, could we move with https://github.com/kubernetes-sigs/kubebuilder/pull/2561 first until it is in the state of getting merged? Then, we get this one merged and re-test all with go 1.18. 

I am looking to avoid breaking the ci and block other PRs. 

/hold

**Comment by camilamacedo86** [comment]
/hold cancel

We can move forward with it now

**Review by camilamacedo86 — APPROVED** [review]
Thank you for the contribution 🥇

**Pull request merged**
